### ISSUE TITLE
Linux build doesn't copy dynamic libraries into output directory

### ISSUE THREAD

**Issue by NQNStudios** [calref/cboe#373]
The linux build currently requires the developer to have the dependencies installed system-wide to *run* the game, not just to build it.
I noticed this because, now that the scons build compiles TGUI for you, the TGUI it links against, won't be installed on your system, because it's built and installed in a local subdirectory. So you get this:

```
run_tests(["build/test/passed"], ["build/bin/boe_test"])
/home/nat/repos/cboe/build/bin/boe_test: error while loading shared libraries: libtgui.so.0.9.5: cannot open shared object file: No such file or directory
os.chdir('/home/nat/repos/cboe')
scons: done building targets.
```

**Comment by CelticMinstrel** [comment]
Probably the simplest solution is to prefer static linking on Linux for most of the libraries.

**Comment by NQNStudios** [comment]
Come to think of it, a portable Linux build would also have to ship the `zenity` executable:

https://github.com/calref/cboe/blob/7cb1770059af94f76dd2d1d6bf27d22c0a714c23/src/tools/winutil.linux.cpp#L137

**Comment by CelticMinstrel** [comment]
That's true – if the goal is to make a standalone Linux build in whatever form (Flatpak for example) then it would need to distribute zenity and any of its dependencies as well.

**Comment by NQNStudios** [comment]
For zenity, I made an issue on their repository to see if anyone can help https://gitlab.gnome.org/GNOME/zenity/-/issues/94

**Comment by NQNStudios** [comment]
I think I could maybe, maybe live with having a Linux build that throws an error message if Zenity isn't installed, saying something to the effect of "OpenBoE for Linux requires Zenity. Please install Zenity through your distribution's package manager."

At least as a stopgap. Linux gaming is so, so, so, niche, I barely get any downloads/purchases of my games' Linux ports so delving into bundling a portable zenity build with cboe seems like a poor use of labor time.

@CelticMinstrel what do you think?

**Comment by NQNStudios** [comment]
Maybe with an open-source project you get more Linux users in the audience, but also--those kinds of users should be pretty comfortable installing an extra thing to make it work, I should think.

**Comment by CelticMinstrel** [comment]
It definitely seems good as a stopgap. But, the main issue to consider is that it'll probably show when they try to save their game. Thus, it _needs_ to be possible to install Zenity without closing BoE and have it work when they try to save again. This suggests that there should be an input box where they can manually type in the path to Zenity, as a last resort in case installing it relies on environment variables that can't be propagated to a running process (seems unlikely, but not impossible).

**Comment by NQNStudios** [comment]
It could happen at launch after querying `which zenity` and force the application to close.

**Comment by NQNStudios** [comment]
> By the way, did you follow up on the response you received to your issue on the Zenity repo?

No, I think if I decide to go that route I'll go to their forum and ask if anyone has compiled a statically linked version & has advice about that.

**Comment by CelticMinstrel** [comment]
> It could happen at launch after querying `which zenity` and force the application to close.

I guess that would work, though it's a little unfortunate they can't even launch the game without installing it.

**Comment by NQNStudios** [comment]
A solution for the "Linux users can't launch the game without Zenity installed" problem would be, have the game manage a folder of save files that can be selected in a simple stack/page interface instead of an OS-specific file browser. Preferences could turn this feature on/off.
